.circleci/build.sh

```diff
@@ -1,26 +1,30 @@
 #!/bin/bash
-
-set -e
+set -eo pipefail
 
 # enables building the test-deployment container with the same script
 # defaults to packages for minimal intervention in the ci config
 FOLDER="${FOLDER:-packages}"
+PUBLISH_IMAGES="${PUBLISH_IMAGES:-false}"
 
-DOCKER_IMAGE_TAG=speckle/speckle-$SPECKLE_SERVER_PACKAGE
+DOCKER_IMAGE_TAG="speckle/speckle-${SPECKLE_SERVER_PACKAGE}"
 
 # IMAGE_VERSION_TAG=$(./.circleci/get_version.sh)
-IMAGE_VERSION_TAG="${IMAGE_VERSION_TAG:-0}"
-echo $IMAGE_VERSION_TAG
+# if there is not image version tag, uses the SHA1 of the last git commit of the branch that triggered this build
+IMAGE_VERSION_TAG="${IMAGE_VERSION_TAG:-${CIRCLE_SHA1}}"
+echo "${IMAGE_VERSION_TAG}"
 
 export DOCKER_BUILDKIT=1
 
-docker build --build-arg SPECKLE_SERVER_VERSION=$IMAGE_VERSION_TAG -t $DOCKER_IMAGE_TAG:latest . -f $FOLDER/$SPECKLE_SERVER_PACKAGE/Dockerfile
+docker build --build-arg SPECKLE_SERVER_VERSION="${IMAGE_VERSION_TAG}" -t "${DOCKER_IMAGE_TAG}:${IMAGE_VERSION_TAG}" . --file "${FOLDER}/${SPECKLE_SERVER_PACKAGE}/Dockerfile"
```

Comment:
we should always have an IMAGE_VERSION_TAG, so build with this unique tag.  Only if we're publishing will we tag it with the more generic `latest` and `2`.